ui/src/app/pages/workspace/workspace-edit.tsx

```diff
@@ -839,30 +839,38 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
       // TODO: This validation spec should include error messages which get
       // surfaced directly. Currently these constraints are entirely separate
       // from the user facing error strings we render.
+      const lengthMessages = (prefix = '') => ({
+        tooShort: `${prefix} cannot be blank`,
+        tooLong: `${prefix} cannot exceed %{count} characters`
+      });
+
       const constraints: object = {
         name: {
-          length: { minimum: 1, maximum: 80 }
+          length: { minimum: 1, maximum: 80, ...lengthMessages('Name')}
         },
         billingAccountName: { presence: true },
-        intendedStudy: { length: { minimum: 1, maximum: 1000 } },
+        intendedStudy: { length: { minimum: 1, maximum: 1000, ...lengthMessages() } },
```

Comment:
To help future maintainers, please add a comment up here indicating why these constraints can be left without a prefix string – e.g. "because it requires HTML formatting, the prefix string is omitted here and included in the React template below".